pkg/agent/cluster.go

```diff
@@ -185,11 +185,14 @@ func (czero *Cluster) IsBootstrapComplete() (bool, error) {
 			return false, errors.New("cluster metadata returned nil from Agent Rest API")
 		}
 
-		if !checkHostsValidations(clusterMetadata, logrus.StandardLogger()) {
-			return false, errors.New("cluster host validations failed")
-		}
-
 		czero.PrintInstallStatus(clusterMetadata)
+
+		// If the status changes back to Ready from Installing it indicates an error
+		if *clusterMetadata.Status == models.ClusterStatusReady &&
+			czero.installHistory.RestAPIPreviousClusterStatus == models.ClusterStatusPreparingForInstallation {
+			logrus.Warn("Cluster has stopped installing... state has changed back to Ready from Preparing")
+			return false, errors.New("cluster installation has changed to Ready from Preparing")
```

Comment:
The error message should be something like "Failed to prepare cluster installation".

Unfortunately, returning an err doesn't actually do anything because the return value is not being checked. We need another mechanism that will result in (a) the error message being printed (instead of the warning above), and (b) the process exiting.

That could mean replacing all of the other errors in this function with nil, since they don't do anything, and then actually checking the return value. Or we could add another return value. Or something else.

Reply:
I changed to return an additional value to indicate whether the process should exit. This allows logs to be output which will not result in an exit, I've also removed some logs (like the validation) that happen every cycle.

When the error occurs the wait-for now exits with:
level=error msg=Bootstrap failed to complete: : bootstrap process returned error: Failed to prepare cluster installation